blendergltf.py

```diff
@@ -874,6 +874,32 @@ def export_gltf(scene_delta, settings={}):
     else:
         mesh_list = scene_delta.get('meshes', [])
 
+    # Use this to store objects that we need to transform back from their
+    # converted axis space. There is no need to copy meshes that don't have any
+    # modifiers applied but that means we have to be careful in modifying them.
+    user_objects = []
+
+    global_matrix = settings.get('global_matrix', None)
+    if global_matrix != None:
+        # If a global matrix was provided apply it to all objects.
+        for obj in object_list:
+            # Find the mesh associated with this object
+            if obj.name in mod_meshes:
+                # Use the modified mesh copy (completely safe to modify).
+                mesh = mod_meshes[obj.name]
+            else:
+                # We have to remember to undo the transformation because this is
+                # a user mesh (not one we created).
+                mesh = obj.data
+                user_objects.append(obj)
+
+            inv_world_mat = obj.matrix_world.inverted()
+
+            # Transform the mesh from model space to blender world space, then
+            # converted-axis world space and finally back to a converted-axis
+            # model space, suitable for copying.
+            mesh.transform(inv_world_mat * global_matrix * obj.matrix_world)
```

Comment:
If multiple objects point to the same mesh, does that mesh get transformed multiple times? I am also not too sure about modifying Blender data. As you have already mentioned, an error could prevent us from undoing the changes. Also, does this end up affecting the undo stack?

An alternative option is to simply apply the global matrix to the node's matrix. However, I could see that messing with game logic. We could also copy all incoming objects so we don't touch the user data. That still has problems with being interrupted, but at least it is just leaves extra meshes around and does not potentially destroy work. Also, I think we would end up with a bunch of screwed up names on the meshes (e.g. adding .001 to the names).

Overall, I am not sure how to best handle this. I think the OBJ exporter has a similar option. It may be worthwhile to see how it, or other exporters, handle it.